openfe_gromacs/protocols/gromacs_md/md_methods.py

```diff
@@ -188,10 +210,158 @@ def get_mdp_filenames(self) -> list[list[pathlib.Path]]:
           list of paths (pathlib.Path) to the mdp files for energy minimization,
           NVT and NPT MD runs
         """
-        mdps = [pus[0].outputs["mdp_files"] for pus in self.data.values()]
+        mdps = [
+            pus[0].outputs["mdp_files"]
+            for pus in self.data.values()
+            if "GromacsMDSetupUnit" in pus[0].source_key
+        ]
 
         return mdps
 
+    def get_gro_em_filename(self) -> list[pathlib.Path]:
+        """
+        Get a list of paths to the .gro file, last frame of the
+        energy minimization
+
+        Returns
+        -------
+        gro : list[pathlib.Path]
+          list of paths (pathlib.Path) to the output .gro file
+        """
+        gro = [
+            pus[0].outputs["gro_em"]
+            for pus in self.data.values()
+            if "GromacsMDRunUnit" in pus[0].source_key
+        ]
+
+        return gro
+
+    def get_gro_em_filename(self) -> list[pathlib.Path]:
+        """
+        Get a list of paths to the .gro file, last frame of the
+        energy minimization
+
+        Returns
+        -------
+        gro : list[pathlib.Path]
+          list of paths (pathlib.Path) to the output .gro file
+        """
+        gro = [
+            pus[0].outputs["gro_em"]
+            for pus in self.data.values()
+            if "GromacsMDRunUnit" in pus[0].source_key
+        ]
+
+        return gro
+
+    def get_tpr_em_filename(self) -> list[pathlib.Path]:
```

Comment:
Instead of getting all filenames separately, get a dictionary of lists of filenames, e.g. `get_filenames_em`. And in addition get gro and xtc separately.

Reply:
Done